### PR TITLE
fix #33721 popup position beeing "fixed" on popup()

### LINKED ISSUE #33721: Popup always "fixes" position when popup() is called
**Godot version:**
v3.2.beta.custom_build (master clone this evening)
**OS/device including version:**
Win 10
**Issue description:**
I was wondering, why 
```
$normal_context.rect_position = event_mouse_position
$normal_context.popup()
```
didn't positioned my popup where I clicked, if I went outside a certain rectangle.
As I tested I found changing the code to
```
$detector/normal_context.popup()
$detector/normal_context.rect_position = event_mouse_position
```
worked. As I went into VS I noticed, that _fix_size() was called, even when the size isn't touched by the function (Bounds were empty)

**Steps to reproduce:**

1. Add Popup as a child of a Node2D
2. Add a camera, which you can move
3. Apply the above code snippet and make sure it is only called once.
4. Move with the camera away
5. Open your popup

### PR DESCRIPTION
**Pull request by HaSa1002**
I don't know if I broke something, but it works in my tests

*Bugsquad edit:* Fixes #33721.

### CODE REVIEW

**Comment by HaSa1002** [comment]
this fix also fixes wrong sizes of the containers when opend through popup()

File: scene/gui/popup.cpp [hunk 1]
```diff
@@ -182,8 +182,8 @@ void Popup::_popup(const Rect2 &p_bounds, const bool p_centered) {
 		} else {
 			set_position(p_bounds.position);
 		}
-	}
 	_fix_size();
```
Comment (akien-mga) [inline scene/gui/popup.cpp]: Indentation issue.

Comment (HaSa1002) [reply scene/gui/popup.cpp]: It was a bit late, yesterday... Fixed this now.

**Comment by akien-mga** [comment]
The thing is that `_fix_size()` has code to ensure that the popup is actually visible and not outside the window, which makes sense to run all the time I guess. I think there are other factors which can impact the position of the popup apart from the optional `p_bounds`, and which would make it meaningful to run the `_fix_size()` logic.

**Comment by HaSa1002** [comment]
Maybe the resetting of the position can be altered wheter the node is a child Node2D or a child of the CanvasItem. Because in the former, it is possible to position it outside of the viewport_rect.

**Comment by YeldhamDev** [comment]
The problem with this commit is that it breaks places where it expects positioned popups to stay inside:
Screenshot_20191119_114253

**Comment by HaSa1002** [comment]
I updated the code now, to run the fixup if it's a child of a control otherwise the fixup will be ignored. That fixes @YeldhamDev problem I created and the bug I intended to fix.

**Comment by YeldhamDev** [comment]
That looks kinda hacky, and you haven't explained why can't you just adjust the position after the popup.

**Comment by akien-mga** [comment]
Superseded by #33936.

**Pull request closed without merging**